File: desktop/src/main/java/imo/common/ImageList.java
package imo.common;


import mine.MineException;
import mine.paint.MineImage;
import mine.paint.MineImageLoader;

public class ImageList {

	public static final String IMAGE_PATH = "imo/image/";

	private MineImage[][] imo;
	private MineImage[][] jiki;
	private MineImage[][] sister;
	private MineImage[][] sword;
	private MineImage turu;
	private MineImage hpbar;

	private MineImage endi;
	
	private MineImageLoader il;

	public ImageList(MineImageLoader il){
		this.il = il;
		imo = il.loadTile(IMAGE_PATH + "imo.png", 32, 32);
		jiki = il.loadTile(IMAGE_PATH + "jiki.png", 32, 32);
		sister = il.loadTile(IMAGE_PATH + "sister.png", 32, 32);
		sword = il.loadTile(IMAGE_PATH + "sword.png", 32, 32);
		turu = il.load(IMAGE_PATH + "turu.png");
		hpbar = il.load(IMAGE_PATH + "hpbar.png");
	}

	public void setEndImage(int n) {
		endi = il.load(IMAGE_PATH + "end" + n + ".png");
	}

	public MineImage getEndImage() {
		return endi;
	}
	/**
	 * @return Returns the hpbar.
	 */
	public MineImage getHpbar() {
		return hpbar;
	}
	/**
	 * @return Returns the imo.
	 */
	public MineImage[][] getImo() {
		return (MineImage[][])imo.clone();
	}
	/**
	 * @return Returns the jiki.
	 */
	public MineImage[][] getJiki() {
		return (MineImage[][])jiki.clone();
	}
	/**
	 * @return Returns the sister.
	 */
	public MineImage[][] getSister() {
		return (MineImage[][])sister.clone();
	}
	/**
	 * @return Returns the sword.
	 */
	public MineImage[][] getSword() {
		return (MineImage[][])sword.clone();
	}
	/**
	 * @return Returns the turu.
	 */
	public MineImage getTuru() {
		return turu;
	}
}

File: core/src/main/java/imo/common/ImageList.kt
package imo.common


import mine.paint.MineImage
import mine.paint.MineImageLoader

class ImageList(private val il: MineImageLoader) {

    val imo: Array<Array<MineImage>>
    val jiki: Array<Array<MineImage>>
    val sister: Array<Array<MineImage>>
    val sword: Array<Array<MineImage>>
    val turu: MineImage
    val hpbar: MineImage

    init {
        imo = il.loadTile(IMAGE_PATH + "imo.png", 32, 32)
        jiki = il.loadTile(IMAGE_PATH + "jiki.png", 32, 32)
        sister = il.loadTile(IMAGE_PATH + "sister.png", 32, 32)
        sword = il.loadTile(IMAGE_PATH + "sword.png", 32, 32)
        turu = il.load(IMAGE_PATH + "turu.png")
        hpbar = il.load(IMAGE_PATH + "hpbar.png")
    }

    fun loadEndImage(n: Int): MineImage {
        return il.load(IMAGE_PATH + "end" + n + ".png")
    }

    companion object {

        const val IMAGE_PATH = "imo/image/"
    }
}
